azavea/cac-tripplanner
# Upgrade to Xenial

## Overview

Upgrade ansible roles to support using Ubuntu Xenial (16.04) as the OS for all three VMs. Mostly, this meant switching services to use `systemd` instead of `upstart`.

This is to support upgrading to python 3 and then Django 2.x later; here, python 2 is still in use.

Also makes some minor upgrades to ansible syntax, to resolve deprecation warnings.


### Notes

I tried using the gunicorn ansible role instead of installing it within the app role, but ran into issues, so instead upgraded the gunicorn role in place for `systemd`. As I've added support for the `timeout` parameter to the gunicorn ansible role, it's possible we could use it in future instead.

For the OTP VM, it was necessary to explicitly install and use python 2, in order to continue to validate GTFS feeds, as the feed validator library does not yet support python 3: google/transitfeed#412

For the python 3 upgrade, we can handle the OTP machine by one of:
 - removing GTFS validation entirely (if using the [feed fetcher library](https://github.com/azavea/gtfs-feed-fetcher), this validation has already happened on download)
 - leaving the machine on python 2 until the validator project upgrades
 - upgrading the validator to python 3 ourselves, which might necessitate forking the project

I would be inclined towards the first option, as the feeds are currently validated outside the VM as part of the current graph update process, and because validating again slows the build process.


## Testing Instructions

 * destroy and rebuild all VMs
 * provisioning should complete as expected
 * no ansible deprecated usage warnings should remain for roles that are defined in this project (external roles still have some warnings, particularly `postgresql` and `postgresql-support`)
 * site should still function as expected
 * `services should start on reboot (site should still work after `vagrant reload`)

Closes #1100 
Closes #995
